src/shared/models/NewRecordsState.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Purpose: Keeps track of IDs that were just created on the backend.
+ * This list gets used to delay network calls to ensure upcoming
+ * requests are ready to be accepted by the backend.
+ */
+export class NewRecordsState {
+  // time in ms
+  OP_REPO_POST_CREATE_DELAY: number;
+  // Key = a string id
+  // Value = A Timestamp in ms of when the id was created
+  private records: Map<string, number> = new Map();
+
+  constructor(time = 3000) {
+    this.OP_REPO_POST_CREATE_DELAY = time;
+  }
+
+  public add(key: string, overwrite = false): void {
```

Comment:
ah good idea on the overwrite logic. Maybe we should add this to the Android SDK as well.

Reply:
Got it from iOS, iOS has this logic: https://github.com/OneSignal/OneSignal-iOS-SDK/pull/1470/files#diff-478e4e86941c7e4be07cbc74b0f7d337844d1c7254ab37ab92a1d50c4da2b006R49